### PR TITLE
[EZ] Update CS Tech Electives & INFO Electives to be 3000+, not 3000 only

### PR DESCRIPTION
**Pull request by willespencer**
### Summary 

Update the CS major technical electives and CS electives checker to include courses that have a course number 3000+ or 4000+ respectively, and not just starting with a 3 or 4. Furthermore, update the INFO Electives requirement to use this new way of checking these courses to be more lenient

- [x] Updated CS technical and CS electives checkers
- [x] Updated Info Electives checker
- [x] Ran req gen

image

Remaining TODOs:

- [ ] This fixes the issue of courses 4000 level or above not being able to be applied properly to technical requirements, but does not fix the more complicated issue of 2000 or below courses that can be selected but should not be. This will be covered in the next sprint.

### Test Plan 

1. Confirm not only 3000 but 4000+ courses can be added to the technical req self check and INFO elective (if INFO classes), and 5000+ CS courses can be added to the CS elective. Ensure that the course shows up under the requirement name and the counter increments for the self check ones.
2. Do the same for courses under 3000 for the technical self check, and confirm the counter does not increment.
3. Confirm that courses that should be exceptions for CS electives and INFO electives are actually exceptions and either do not count when added to, or are not allowed to be added to. Previous bugs in the code seem to have missed some exceptions.

### CODE REVIEW

**Review by SamChou19815 — APPROVED** [review]

**Review by tcho6319 — CHANGES_REQUESTED** [review]
Thanks Will! Works according to the updated checker but I pointed out how you'd want to handle an edge case. Once the comment is addressed, this PR should be g2g.

File: src/requirements/data/majors/cs.ts [hunk 1]
```diff
@@ -84,7 +84,12 @@ const csRequirements: readonly CollegeOrMajorRequirement[] = [
     checker: [
       (course: Course): boolean => {
         const { catalogNbr } = course;
-        return catalogNbr.startsWith('3');
+        return (
+          ifCodeMatch(catalogNbr, '3***') ||
+          ifCodeMatch(catalogNbr, '4***') ||
+          ifCodeMatch(catalogNbr, '5***') ||
+          ifCodeMatch(catalogNbr, '6***')
+        );
```
[the remark this answers is not in the document]
Comment (willespencer) [reply src/requirements/data/majors/cs.ts:92]: I could do that! I was originally considering that, but then I noticed the `Electives` requirement in the Info major checks 3000+ in this way. Should I change both to match your suggestion?

I also just noticed `CS Electives` seems to be the same as technical, in which it only checks 4000, but it should be 4000+. I'll also change it to match whatever I do for Technical and Info Electives!

[the remark this answers is not in the document]
Comment (tcho6319) [reply src/requirements/data/majors/cs.ts:92]: I think by doing it the way I suggested, we would air on the side of allowing more courses (i.e. being more lenient with our checker) than accidentally leaving out certain levels.

[the remark this answers is not in the document]
Comment (willespencer) [reply src/requirements/data/majors/cs.ts]: Agreed! Just updated the code for CS technical electives, and applied the logic to CS 4000+ electives and INFO electives

**Review by tcho6319 — APPROVED** [review]
LGTM. Thanks Will! 

1. I see a similar issue on master as well so I don't think this issue is from any changes you made, but Technical Electives doesn't appear as a valid requirement when trying to add CS 7090 from the add modal. The assignment works when adding CS 7090 from the self check add modal. I will make a Notion item, but I wanted to lyk!

https://user-images.githubusercontent.com/54298311/111922000-fa47a600-8a6d-11eb-9ca0-4656e35585f9.mov

**Pull request merged**